packages/headless/src/features/search/search-analytics-actions.ts

```diff
@@ -1,7 +1,30 @@
+import {SearchAPIErrorWithStatusCode} from '../../api/search/search-api-error-response';
+import {getAdvancedSearchQueriesInitialState} from '../advanced-search-queries/advanced-search-queries-state';
 import {makeAnalyticsAction, AnalyticsType} from '../analytics/analytics-utils';
+import {getQueryInitialState} from '../query/query-state';
 
 export const logFetchMoreResults = makeAnalyticsAction(
   'search/logFetchMoreResults',
   AnalyticsType.Search,
   (client) => client.logFetchMoreResults()
 );
+
+export const logQueryError = (error: SearchAPIErrorWithStatusCode) =>
+  makeAnalyticsAction(
+    'search/queryError',
+    AnalyticsType.Search,
+    (client, state) =>
+      client.logQueryError({
+        query: state.query?.q || getQueryInitialState().q,
+        aq:
+          state.advancedSearchQueries?.aq ||
+          getAdvancedSearchQueriesInitialState().aq,
```

Comment:
I don't know how critical it is, but if the first half these conditions is `undefined`, it is because the user is not using that feature/slice reducer.

In these cases, should we send the initial state (empty string)? or `null`/`undefined` or signify that "there is no aq", rather than "there is an aq set to an empty string"? or another option might be to not send the param at all?

For example, in the search request, when a reducer is not present, we do not send the parameter.

Reply:
The query error event in the analytics library does not accept undefined as a value, I assumed that means we have to send empty values https://github.com/coveo/coveo.analytics.js/commit/c499a2a005af440fce903e923b1e3a08c6be3101#diff-85ba150248ab396b6ffd753401adecf490b03715f724a0940ab4dbb149d58a17R237

Most of the time they'll be empty